### ISSUE TITLE
Implement /trace_route

### ISSUE THREAD

**Issue by nilsnolde** [gis-ops/valhalla-app#159]
### What would you like to share?

Extend the frontend with map-matching functionalities. I'd suggest to support both inputs, encoded polyline (e.g. from an existing Valhalla response `shape`) and GPX (will need conversion to JSON, incl timestamps ideally). There'd obviously be no waypoint input by the user. 

Approx:
- Users can "upload" a GPX file (needs max file size) or copy/paste an encoded polyline
- We parse the polyline or GPX file and display the input to the user simply as a LineString
- When the user chooses to map-match with the input, hits a button
- We send the input to `/trace_route` and generate the maneuvers & route summary and of course display the final route

Devil's in the details of course, but might not be that crazy much work to realize.

**Comment by MihirKp25** [comment]
Out of context from issue. Is there any way we can create a GPX file from OSM only?

**Comment by nilsnolde** [comment]
Yeah, more one for stackoverflow or OSM forum. But also not sure I understand the question.. GPX is just another „geographic“ format for some random LineString. OSM has nothing to do with it. Usually you’d record a track via GPS and the exchange/export format is (unfortunately) often GPX. Nothing stops you from creating a custom GPX file manually or via some scripting.
